cogs/Misc.py

```diff
@@ -1,36 +1,151 @@
+import csv
+import io
+import os
+import sqlite3
+from dataclasses import dataclass
+from typing import Iterable
+
 import discord
+from discord.commands.context import ApplicationContext
 from discord.ext import commands
+from discord.interactions import Interaction
+
+DATABASE_FILES = [
+    "util/database.sqlite",
+    "bandwidth.db",
+]
+LOADING_EMOJI = "\N{Clockwise Downwards and Upwards Open Circle Arrows}"
+COMPLETED_EMOJI = "\N{White Heavy Check Mark}"
+DUMP_GUILD = os.getenv("ALLOW_DUMP_DATABASE_GUILD")
+DUMP_ROLE = os.getenv("ALLOW_DUMP_DATABASE_ROLE")
+
+DUMP_PERMISSIONS = (lambda x: x) if DUMP_ROLE is None else commands.has_role(DUMP_ROLE)
+
+
+@dataclass
+class Table:
+    database_file_path: str
+    database_name: str
+    name: str
+    column_names: list[str]
+
+
+def grep_tables() -> list[Table]:
+    tables: list[Table] = []
+
+    for database_file in DATABASE_FILES:
+        database_name = database_file.rsplit("/", maxsplit=1)[-1].split(".")[0]
+
+        with sqlite3.connect(database_file) as conn:
+            cursor = conn.cursor()
+            cursor.execute(
+                """
+                SELECT 
+                    name
+                FROM 
+                    sqlite_schema
+                WHERE 
+                    type ='table'
+                    AND 
+                    name NOT LIKE 'sqlite_%'
+                """
+            )
+
+            for (table_name,) in cursor.fetchall():
+                cursor.execute(f"PRAGMA table_info({table_name})")
+                tables.append(
+                    Table(
+                        database_file_path=database_file,
+                        database_name=database_name,
+                        name=table_name,
+                        column_names=[x[1] for x in cursor.fetchall()],
+                    )
+                )
+
+    return tables
+
+
+def dump_tables_to_csv(tables: list[Table]) -> Iterable[discord.File]:
+    for table in tables:
+        buffer = io.StringIO()
+        writer = csv.writer(buffer)
+        writer.writerow(table.column_names)
+
+        with sqlite3.connect(table.database_file_path) as conn:
+            cursor = conn.cursor()
+            cursor.execute(f"SELECT * FROM {table.name}")
```

Comment:
CodeFactor Error may be ignored, SQLi not possible (unless someone manages to change table names)